javaslang/src/main/java/javaslang/collection/Stream.java

```diff
@@ -418,6 +418,76 @@ public T next() {
         return Stream.ofAll(Iterator.ofAll(array));
     }
 
+    /**
+     * Wrap a <code>InputStream</code> into a <code>Stream</code> of bytes.
+     * <p>
+     * Client code must close the <code>InputStream</code>. All
+     * {@link IOException}'s thrown be the <code>InputStream</code> are wrapped
+     * by {@link UncheckedIOException}'s.
+     */
+    static Stream<Byte> in(InputStream in) {
+        return StreamFactory.create(new AbstractIterator<Byte>() {
+            Byte next = null;
+
+            @Override
+            protected Byte getNext() {
+                Byte result = next;
+                next = null;
+                return result;
+            }
+
+            @Override
+            public boolean hasNext() {
+                if (next == null) {
+                    try {
+                        int i = in.read();
+                        if (i >= 0) {
+                            next = (byte) i;
+                        }
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(e);
```

Comment:
I'm not comfortable with `hasNext` throwing, instead of returning false in this case, since that's why I'm calling it :/